core/trino-parser/src/main/java/io/trino/sql/tree/Query.java

```diff
@@ -133,29 +133,30 @@ public <R, C> R accept(AstVisitor<R, C> visitor, C context)
     @Override
     public List<Node> getChildren()
     {
-        ImmutableList.Builder<Node> nodes = ImmutableList.builder();
-        nodes.addAll(functions);
-        nodes.addAll(sessionProperties);
-        with.ifPresent(nodes::add);
-        nodes.add(queryBody);
-        orderBy.ifPresent(nodes::add);
-        offset.ifPresent(nodes::add);
-        limit.ifPresent(nodes::add);
-        return nodes.build();
+        return ImmutableList.<Node>builder()
+                .addAll(functions)
+                .addAll(sessionProperties)
+                .addAll(with.stream().toList())
```

Comment:
it can be also: `with.map(ImmutableList::of).orElseGet(ImmutableList::of)`

Reply:
Yeah, but that's harder to read, and I doubt it makes a difference in practice.